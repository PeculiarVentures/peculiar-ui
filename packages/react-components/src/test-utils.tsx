import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
