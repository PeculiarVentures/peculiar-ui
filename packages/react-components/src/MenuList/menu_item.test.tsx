import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { MenuItem } from './index';

describe('<MenuItem />', () => {
  it('should render with default styles', () => {
    const { asFragment } = render(<MenuItem>MenuItem</MenuItem>);

    expect(asFragment()).toMatchSnapshot();
  });
});
