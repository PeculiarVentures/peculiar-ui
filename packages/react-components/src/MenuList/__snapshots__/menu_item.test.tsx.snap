// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItem /> should render with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: 0px var(--pv-size-base-2);
  font-family: inherit;
  outline: none;
  width: 100%;
  height: var(--pv-size-base-7);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  -webkit-transition: background-color 200ms;
  transition: background-color 200ms;
  background-color: transparent;
  border: none;
  color: var(--pv-color-black);
  box-sizing: border-box;
  gap: var(--pv-size-base-2);
}

.emotion-0:hover {
  background-color: var(--pv-color-gray-3);
}

.emotion-0:focus {
  background-color: var(--pv-color-gray-4);
}

.emotion-0:active {
  background-color: var(--pv-color-gray-5);
}

.emotion-1 {
  margin: 0;
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  color: inherit;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

<li
    class="emotion-0"
    role="menuitem"
    tabindex="-1"
  >
    <span
      class="emotion-1"
    >
      MenuItem
    </span>
  </li>
</DocumentFragment>
`;
